internal/error.go

```diff
@@ -303,6 +317,27 @@ func NewApplicationError(msg string, errType string, nonRetryable bool, cause er
 	return applicationErr
 }
 
+func NewApplicationErrorWithOptions(msg string, errType string, attributes ApplicationErrorOptions) error {
```

Comment:
Can this return `*ApplicationError` and follow the recommended Go practice?

Reply:
Could you kindly share the practices you have in mind?

Reply:
It is often better Go practice to return the concrete structure (w/ pointer as needed, such as here) instead of the interface (`error`). But we haven't followed that in `NewApplicationError` or `NewApplicationErrorWithCause` or any other `NewXError` call in the `temporal` package, so I can understand both arguments. I do not have an opinion and could totally understand leaving it as is for consistency.